odl/discr/lp_discr.py

```diff
@@ -221,18 +229,88 @@ def element(self, inp=None):
                         ''.format(arr.shape, self.shape))
             arr = arr.ravel(order=self.order)
             return self.element_type(self, self.dspace.element(arr))
-        except TypeError as err:
+        except (TypeError, IndexError) as err:
             if str(err.args[0]).startswith('output contains points outside'):
                 raise err
             else:
                 raise_from(TypeError('unable to create an element of {} from '
-                                     '{!r}.'.format(self, inp)), err)
+                                     '{!r}: {}'.format(self, inp, err)), err)
 
     @property
     def interp(self):
         """Interpolation type of this discretization."""
         return self._interp
 
+    def as_complex_space(self, dtype=None):
```

Comment:
Not quite sure i want different functions here. I'd prefer one big `astype` that takes `dtype`, `impl` etc.